python/ray/workflow/workflow_storage.py

```diff
@@ -86,6 +87,124 @@ def is_recoverable(self) -> bool:
         )
 
 
+class WorkflowIndexingStorage:
+    """Access and maintenance the indexing of workflow status.
+
+    It runs a protocol that guarantees we can recover from any interrupted
+    status updating. This protocol is **not thread-safe** for a workflow, currently
```

Comment:
How about get_status in api.py? It seems that it's accessing the storage directly.

Reply:
this would potentially cause some inconsistency. E.g. the dirty mark is removed by each other when updating the status at the same time.